vector/src/main/java/im/vector/app/features/location/live/map/GetListOfUserLiveLocationUseCase.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location.live.map
+
+import androidx.lifecycle.asFlow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.mapLatest
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.getRoom
+import javax.inject.Inject
+
+class GetListOfUserLiveLocationUseCase @Inject constructor(
+        private val session: Session,
```

Comment:
If I understood well, this is what is called when injecting the `Session`: https://github.com/vector-im/element-android/blob/955f366690def07fa90877cb49224ca7cbab0f99/vector/src/main/java/im/vector/app/core/di/SingletonModule.kt#L136 Am I right?

Reply:
with a quick search I can see that we're injecting the `Session` into lots of places, not a blocking comment :+1: 

the main reason to avoid injecting the `Session` directly is that it means the class won't be able to support multi sessions in the future (although perhaps we should have a Session wrapper which delegates to the active session rather than a separate active session holder...)